frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1611,6 +1611,49 @@ function ReaderHighlight:getHighlightBookmarkItem()
     end
 end
 
+function ReaderHighlight:_insertHighlight(page, highlight)
+    local position
+    if self.view.highlight.saved[page] == nil or #self.view.highlight.saved[page] == 0 then
+        position = 1
+    else
+        local page_highlights = self.view.highlight.saved[page]
+        local highlight_beginning = highlight.pos0
+        local highlight_end = highlight.pos1
+        if self.ui.rolling then
+            for i, cur_highlight in ipairs(page_highlights) do
+                local order = self.ui.document:compareXPointers(highlight_beginning, cur_highlight.pos0)
+                if order == 0 then
+                    order = self.ui.document:compareXPointers(highlight_end, cur_highlight.pos1)
+                end
+                if order > 0 then
+                    position = i
+                    break
+                end
+            end
+        else
+            highlight_beginning.page = page
+            highlight_end.page = page
+            for i, cur_highlight in ipairs(page_highlights) do
+                local cur_pos, cur_pos_end = cur_highlight.pos0, cur_highlight.pos1
```

Comment:
Please note that pos0 and pos1 can be not sorted in pdf, depending on the finger highlighting direction.

Reply:
Is it only for PDFs or all documents.

Reply:
Yes, pdf only.
Besides, native positions must be compared (reflow mode off temporarily).

Reply:
Thanks. Done the necessary changes.